docs/user-guide/at-tls-configuration.md

```diff
@@ -1,23 +1,670 @@
+
+The Discovery Service endpoints are not reachable by standard API Gateway routing by default.
+
+Zowe v3 includes a new component named ZAAS (Zowe Authentication and Authorization Service). In AT-TLS-aware mode, calls to this service must not include X.509 Client Certificate.
+:::
+
+### Limitations
+
+If using AT-TLS with a z/OS Keyring backed by an ICSF hardware module, the only supported configuration is Zowe with z/OSMF authentication provider in JWT mode.
+LTPA token and SAF provider cannot be used in this configuration because API ML cannot access the hardware key to sign its own tokens.
+Personal Access Tokens (PAT) are not supported in this configuration because API ML cannot access the hardware key to sign the tokens.
+
+## AT-TLS rules
+
+This section describes suggested AT-TLS settings, and serves as guidelines to set your AT-TLS rules.
+
+### Inbound rules
+
+A generic inbound rule can be set for all Zowe services:
+
+```bash
+TTLSRule ZoweServerRule
+{
+  LocalAddr All
+  RemoteAddr All
+  LocalPortRange 7551-7559 # Range covers all Zowe services
+  Jobname ZWE1* # Jobname according to zowe.job.prefix in zowe.yaml
+  Direction Inbound
+  TTLSGroupActionRef ServerGroupAction
+  TTLSEnvironmentActionRef ZoweServerEnvironmentAction
+  TTLSConnectionActionRef ZoweServerConnectionAction
+}
+
+TTLSGroupAction ServerGroupAction
+{
+  TTLSEnabled On
+}
+
+TTLSEnvironmentAction ZoweServerEnvironmentAction
+{
+  HandshakeRole ServerWithClientAuth
+  EnvironmentUserInstance 0
+  TTLSEnvironmentAdvancedParmsRef ServerEnvironmentAdvParms
+  TTLSKeyringParmsRef ZoweKeyring
+}
+
+TTLSConnectionAction ZoweServerConnectionAction
+{
+  HandshakeRole ServerWithClientAuth
+  TTLSCipherParmsRef CipherParms
+  TTLSConnectionAdvancedParmsRef ZoweConnectionAdvParms
+}
+```
+
+The `PortRange` of this inbound rule is taken from the list of API Mediation Layer components in the `zowe.yaml` file. The `PortRange` should cover the following components:
+
+| Component | Default Port |
+|----|-----------------------|
+| Gateway | 7554 |
+| Discovery | 7553 |
+| Caching Service | 7555 |
+| API Catalog | 7552 |
+| ZAAS | 7558 |
+| Zowe System Services (ZSS) | 7557 |
+| Zowe Application Server | 7556 |
+
+__Follow this step:__
+
+Replace `ZoweKeyring` with the keyring configured for your installation. Follow [the SAF keyring instructions](../../getting-started/zowe-certificates-overview.md#saf-keyring) in the article _Zowe Certificates overview_ to configure keyrings for your Zowe instance.
+
+Note the setting `HandshakeRole`. This setting applies to core services which authenticate through certificates with each other. This setting allows the API Gateway to receive and accept X.509 client certificates from API Clients.
+
+For more granularity in the AT-TLS rules, separate the rules that need to support Client Certificate authentication (Discovery Service, Gateway Service) from the rest.
```

Comment:
Can you please describe with a bit more specificity what you mean by "the rest"?

Reply:
I copy the text from the updated v2 version